ports/stm32f4/common-hal/busio/UART.c

```diff
@@ -567,33 +568,53 @@ STATIC void uart_clock_disable(uint16_t mask) {
 
 STATIC void uart_assign_irq(busio_uart_obj_t *self, USART_TypeDef * USARTx) {
     #ifdef USART1
```

Comment:
Maybe make this a table lookup or a `switch`, still with appropriate `#ifdef`'s.

Reply:
I guess the switch would be like:
```
switch (USARTx)
{
#ifdef USART1
    case USART1: self->irq = USART1_IRQn;
        break;
#endif
#ifdef USART2
//etc
    default: // code to be executed if n doesn't match any cases
}
```
But how would a table lookup work without extractable index information? And would the reason for one of these be just code line reduction, or is it something else?

Reply:
The table would be like:
```
{ USART1, USART1_IRQn },
{ USART2, USART2_IRQn },
```
etc.
I think a `switch` is fine. The main thing is that the current structure has each if statement fall into the next one. If the compiler is clever it won't be extra code. You could add `else if` instead. But I think a `switch` is cleaner.

Reply:
Whoops, can't switch on pointer values.

Reply:
The table's going to be kind of ugly too, since I'll either need to assign an empty row for every undefined category or have the size be dynamic. Are you set on me changing this?

Reply:
ok, never mind, not a big deal